Detectors/EMCAL/reconstruction/include/EMCALReconstruction/CaloRawFitter.h

```diff
@@ -0,0 +1,156 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \class EMCalRawFitter
+///
+/// \brief  Base class for extraction of signal amplitude and peak position
+/// Base class for extraction
+/// of signal amplitude and peak position
+/// from CALO Calorimeter RAW data.
+/// It contains some utilities for preparing / selecting
+/// signals suitable for signal extraction
+///
+/// \author Hadi Hassan <hadi.hassan@cern.ch>, Oak Ridge National Laboratory
+/// \since November 4th, 2019
+///
+
+#ifndef EMCALRAWFITTER_H_
+#define EMCALRAWFITTER_H_
+
+#include <iosfwd>
+#include <array>
+#include <Rtypes.h>
+#include "EMCALReconstruction/CaloFitResults.h"
+#include "DataFormatsEMCAL/Constants.h"
+#include "EMCALReconstruction/AltroDecoder.h"
+
+namespace o2
+{
+
+namespace emcal
+{
+
+class CaloRawFitter
+{
+
+ public:
+  /// \brief Constructor
+  CaloRawFitter(const char* name, const char* nameshort);
+
+  /// \brief Destructor
+  ~CaloRawFitter() = default;
+
+  /// \brief Method to do the selection of what should possibly be fitted.
+  /// \return Size of the sub-selected sample,
+  /// \return index of the bunch with maximum signal,
+  /// \return maximum signal,
+  /// \return maximum aplitude,
+  /// \return index of max aplitude in array,
+  /// \return pedestal,
+  /// \return first time bin,
+  /// \return last time bin,
+  /// \return ampitude cut
+  std::tuple<int, int, float, short, short, float, int, int> preFitEvaluateSamples(const std::vector<AltroDecoder::Bunch>& bunchvector,
+                                                                                   std::optional<unsigned int> altrocfg1, std::optional<unsigned int> altrocfg2, int acut);
+
+  /// \brief The require time range if the maximum ADC value is between min and max (timebin)
+  void setTimeConstraint(int min, int max);
+
+  void setIsZeroSuppressed(bool iszs = true) { mIsZerosupressed = iszs; }
+  void setAmpCut(float cut) { mAmpCut = cut; }
+  void setNsamplePed(int i) { mNsamplePed = i; }
+  void setL1Phase(double phase) { mL1Phase = phase; }
+
+  bool getIsZeroSuppressed() const { return mIsZerosupressed; }
+  float getAmpCut() const { return mAmpCut; }
+  int getNsamplePed() const { return mNsamplePed; }
+
+  // access to array info
+  double getReversed(const int i) const { return mReversed[i]; }
+  const char* getAlgoName() const { return mName.c_str(); }
+  const char* getAlgoAbbr() const { return mNameShort.c_str(); }
+  FitAlgorithm getAlgo() const { return mAlgo; }
```

Comment:
FitAlgorithm is not defined anywhere.

Reply:
It should be defined in DataFormatsEMCAL/Constants.h. It was eventually forgotten in the commit.